### PR TITLE
Always mount istio certs lazily

### LINKED ISSUE #1193: Always mount istio auth certs to enable dynamic auth flag
Please mount istio certs with "optional" flag to allow auth feature flag without pod restarts.

### PR DESCRIPTION
**Pull request by kyessenov**
**What this PR does / why we need it**:
Reduces variation in the templates to lazily mount when they are available.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1193

**Special notes for your reviewer**:
The goal is to avoid re-deployment when turning on auth in Istio.

**Release note**:

```release-note
lazily mount mtTLS certs for Istio Auth
```

### CODE REVIEW

File: platform/kube/inject/testdata/first-initializer.yaml.patch [hunk 1]
```diff
@@ -1 +1 @@
...
...tainers":[{"name":"hello"},{"name":"istio-proxy"}],"containers":[{"args":["proxy","sidecar","-v","2"],"env":[{"name":"POD_NAME","valueFrom":{"fieldRef":{"fieldPath":"metadata.name"}}},{"name":"POD_NAMESPACE","valueFrom":{"fieldRef":{"fieldPath":"metadata.namespace"}}},{"name":"INSTANCE_IP","valueFrom":{"fieldRef":{"fieldPath":"status.podIP"}}}],"image":"docker.io/istio/proxy_debug:unittest","imagePullPolicy":"IfNotPresent","name":"istio-proxy","resources":{},"securityContext":{"readOnlyRootFilesystem":true,"runAsUser":1337},"volumeMounts":[{"mountPath":"/etc/istio/config","name":"istio-config","readOnly":true},{"mountPath":"/etc/istio/proxy","name":"istio-envoy"},{"mountPath":"/etc/certs","name":"istio-certs","readOnly":true}]}],"initContainers":[{"args":["-p","15001","-u","1337"],"image":"docker.io/istio/proxy_init:unittest","imagePullPolicy":"IfNotPresent","name":"istio-init","resources":{},"securityContext":{"capabilities":{"add":["NET_ADMIN"]},"privileged":true}}],"volumes":[{"configMap":{"name":"istio"},"name":"istio-config"},{"emptyDir":{"medium":"Memory","sizeLimit":"0"},"name":"istio-envoy"},{"name":"istio-certs","secret":{"optional":true,"secretName":"istio.default"}}]}}}}
```
Comment (rkpagadala) [inline platform/kube/inject/testdata/first-initializer.yaml.patch]: this is impossible to read and understand, can we change this to multi line json so diffs make more sense.

File: platform/kube/inject/testdata/required.yaml.patch [hunk 2]
```diff
@@ -1 +1 @@
...
...tainers":[{"name":"hello"},{"name":"istio-proxy"}],"containers":[{"args":["proxy","sidecar","-v","2"],"env":[{"name":"POD_NAME","valueFrom":{"fieldRef":{"fieldPath":"metadata.name"}}},{"name":"POD_NAMESPACE","valueFrom":{"fieldRef":{"fieldPath":"metadata.namespace"}}},{"name":"INSTANCE_IP","valueFrom":{"fieldRef":{"fieldPath":"status.podIP"}}}],"image":"docker.io/istio/proxy_debug:unittest","imagePullPolicy":"IfNotPresent","name":"istio-proxy","resources":{},"securityContext":{"readOnlyRootFilesystem":true,"runAsUser":1337},"volumeMounts":[{"mountPath":"/etc/istio/config","name":"istio-config","readOnly":true},{"mountPath":"/etc/istio/proxy","name":"istio-envoy"},{"mountPath":"/etc/certs","name":"istio-certs","readOnly":true}]}],"initContainers":[{"args":["-p","15001","-u","1337"],"image":"docker.io/istio/proxy_init:unittest","imagePullPolicy":"IfNotPresent","name":"istio-init","resources":{},"securityContext":{"capabilities":{"add":["NET_ADMIN"]},"privileged":true}}],"volumes":[{"configMap":{"name":"istio"},"name":"istio-config"},{"emptyDir":{"medium":"Memory","sizeLimit":"0"},"name":"istio-envoy"},{"name":"istio-certs","secret":{"optional":true,"secretName":"istio.default"}}]}}}}
```
Comment (rkpagadala) [inline platform/kube/inject/testdata/required.yaml.patch]: this is impossible to read and understand, can we change this to multi line json so diffs make more sense.

**Comment by rshriram** [comment]
if the intent is to just reload with auth, shouldn't the tests change as well?

File: platform/kube/inject/testdata/first-initializer.yaml.patch (hunk 1 above)
Comment (kyessenov) [reply platform/kube/inject/testdata/first-initializer.yaml.patch]: I think the serializer does what it does without prettifying the output. @ayj correct me if I'm wrong.

**Comment by kyessenov** [comment]
That would be the final PR. Right now, i'm trying to decouple mesh config from the agent/kube inject.

File: platform/kube/inject/testdata/first-initializer.yaml.patch (hunk 1 above)
Comment (ayj) [reply platform/kube/inject/testdata/first-initializer.yaml.patch]: Correct - this is the expected JSON output of [strategicpatch.CreateTwoWayMergePatch()](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/strategicpatch/patch.go#L94) which is passed to k8s go-client's PATCH method.

**Comment by ldemailly** [comment]
/cry about conflicts with #1202 :-) I guess I rerun the golden and all should be swell

**Pull request merged**